addons/hr_homeworking_calendar/static/src/calendar/common/badge_selection_icons_mapped_field/badge_selection_icons_mapped_field.js

```diff
@@ -0,0 +1,41 @@
+import { registry } from "../../../../../../web/static/src/core/registry";
+import {
+    badgeSelectionWithIconsField,
+    BadgeSelectionWithIconsField
+} from "../../../../../../mail/static/src/views/fields/badge_selection_icons/badge_selection_icons_field";
+import { useSpecialData } from "../../../../../../web/static/src/views/fields/relational_utils";
+
+
+export class BadgeSelectionWithIconsMappedField extends BadgeSelectionWithIconsField {
+    static props = {
+        ...BadgeSelectionWithIconsField.props,
+    }
+
+    async setup() {
+        const map = {
+            'office': 'fa-building-o',
+            'home': 'fa-home',
+        }
+        const fallback = 'fa-map-marker';
```

Comment:
Actually I remember adding something like that. Check `selection_badge_icon_mapping` in appointment. You can probably move that code in calendar in a separate commit so that you can use it here (JS + static template) (as both modules depend on appointment + only JS so no migration script)

Reply:
If you really need a different 'default icon' you may add it as a prop on the existing component